avocado/utils/nfs.py

```diff
@@ -0,0 +1,349 @@
+"""
+Basic nfs support for Linux host. It can support the remote
+nfs mount and the local nfs set up and mount.
+"""
+import re
+import os
+import logging
+import commands
+
+from . import path
+from . import process
+from . import service
+
+from utils_conn import SSHConnection
+import utils_misc
+
+
+def nfs_exported():
+    """
+    Get the list for nfs file system already exported
+
+    :return: a list of nfs that is already exported in system
+    :rtype: a lit of nfs file system exported
+    """
+    exportfs = process.system_output("exportfs -v")
+    if not exportfs:
+        return {}
+
+    nfs_exported_dict = {}
+    for fs_info in re.findall(r"[/\w+]+.*?\(.*?\)", exportfs, re.S):
+        fs_info = fs_info.strip().split()
+        if len(fs_info) == 2:
+            nfs_src = fs_info[0]
+            access_ip = re.findall(r"(.*)\(", fs_info[1])[0]
+            if "world" in access_ip:
+                access_ip = "*"
+            nfs_tag = "%s_%s" % (nfs_src, access_ip)
+            permission = re.findall(r"\((.*)\)", fs_info[1])[0]
+            nfs_exported_dict[nfs_tag] = permission
+
+    return nfs_exported_dict
+
+
+class Exportfs(object):
+
+    """
+    Add or remove one entry to exported nfs file system.
+    """
+
+    def __init__(self, fs_path, client="*", options="", ori_exported=None):
+        if ori_exported is None:
+            ori_exported = []
+        self.path = fs_path
+        self.client = client
+        self.options = options.split(",")
+        self.ori_exported = ori_exported
+        self.entry_tag = "%s_%s" % (self.path, self.client)
+        self.already_exported = False
+        self.ori_options = ""
+
+    def is_exported(self):
+        """
+        Check if the directory is already exported.
+
+        :return: If the entry is exported
+        :rtype: Boolean
+        """
+        ori_exported = self.ori_exported or nfs_exported()
+        if self.entry_tag in ori_exported.keys():
```

Comment:
nitpick: the `{}.keys()` produces a copy of the dict's keys. For this kind of checks (or for iterations in general) it's recommended to use `{}.iterkeys()` which simply goes through the original set of keys, which is faster and less memory intense.